tests/integration/data/test_namespace.py

```diff
@@ -0,0 +1,159 @@
+import os
+import time
+
+import pytest
+
+from pinecone.core.openapi.db_data.models import NamespaceDescription
+
+
+def setup_namespace_data(index, namespace: str, num_vectors: int = 2):
+    """Helper function to set up test data in a namespace"""
+    vectors = [(f"id_{i}", [0.1, 0.2]) for i in range(num_vectors)]
+    index.upsert(vectors=vectors, namespace=namespace)
+    # Wait for data to be upserted
+    time.sleep(5)
+
+
+def verify_namespace_exists(index, namespace: str) -> bool:
+    """Helper function to verify if a namespace exists"""
+    try:
+        index.describe_namespace(namespace)
+        return True
+    except Exception:
+        return False
+
+
+def delete_all_namespaces(index):
+    """Helper function to delete all namespaces in an index"""
+    try:
+        # Get all namespaces
+        namespaces = list(index.list_namespaces())
+
+        # Delete each namespace
+        for namespace in namespaces:
+            try:
+                index.delete_namespace(namespace.name)
+            except Exception as e:
+                print(f"Error deleting namespace {namespace.name}: {e}")
+
+        # Wait for deletions to complete
+        time.sleep(5)
+    except Exception as e:
+        print(f"Error in delete_all_namespaces: {e}")
+
+@pytest.mark.skipif(
```

Comment:
Sort of nitpicky, but I think instead of doing a `skipif` check and worrying about an env variable, you could just skip always and then update in the gRPC pr:

```suggestion
@pytest.mark.skip(reason="enable when gRPC implemented")
```